src/overlays/gamestates/ovl_select/z_select.c

```diff
@@ -1,40 +1,934 @@
 /*
  * File: z_select.c
  * Overlay: ovl_select
- * Description:
+ * Description: Debug map select
  */
 
 #include "z_select.h"
+#include "alloca.h"
+#include "overlays/gamestates/ovl_title/z_title.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/D_80802390.s")
+void Select_LoadTitle(SelectContext* this) {
+    {
+        GameState* state = &this->state;
+        state->running = false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/func_80800910.s")
+    SET_NEXT_GAMESTATE(&this->state, Title_Init, TitleContext);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/func_80800930.s")
+void Select_LoadGame(SelectContext* this, u32 entranceIndex, s32 opt) {
+    if (gSaveContext.fileNum == 0xFF) {
+        Sram_InitDebugSave();
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/func_80800A44.s")
+    gSaveContext.buttonStatus[0] = BTN_ENABLED;
+    gSaveContext.buttonStatus[1] = BTN_ENABLED;
+    gSaveContext.buttonStatus[2] = BTN_ENABLED;
+    gSaveContext.buttonStatus[3] = BTN_ENABLED;
+    gSaveContext.buttonStatus[4] = BTN_ENABLED;
+    gSaveContext.unk_3F1E = 0;
+    gSaveContext.unk_3F20 = 0;
+    gSaveContext.unk_3F22 = 0;
+    gSaveContext.unk_3F24 = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/func_808013B8.s")
+    Audio_QueueSeqCmd(NA_BGM_STOP);
+    gSaveContext.entranceIndex = entranceIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/D_8080343C.s")
+    if (opt != 0) {
+        gSaveContext.entranceIndex =
+            Entrance_CreateIndex((s32)gSaveContext.entranceIndex >> 9, opt, gSaveContext.entranceIndex & 0xF);
+    }
+    if (gSaveContext.entranceIndex == 0xC000) {
+        gSaveContext.day = 0;
+        gSaveContext.time = CLOCK_TIME(6, 0) - 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/func_80801594.s")
+    gSaveContext.respawn[0].entranceIndex = 0xFFFF;
+    gSaveContext.seqIndex = 0xFF;
+    gSaveContext.nightSeqIndex = 0xFF;
+    gSaveContext.showTitleCard = true;
+    gSaveContext.respawnFlag = 0;
+    gSaveContext.respawn[4].entranceIndex = 0xFF;
+    gSaveContext.respawn[5].entranceIndex = 0xFF;
+    gSaveContext.respawn[6].entranceIndex = 0xFF;
+    gSaveContext.respawn[7].entranceIndex = 0xFF;
+    D_801BDBB0 = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_select/D_80803588.s")
+    do {
+        GameState* state = &this->state;
+        state->running = false;
+    } while (0);
```

Comment:
Is there any way to change this into a single `if (1) {}`?

Reply:
I tried a few times. Nothing else seems to fix it